google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java

```diff
@@ -257,22 +259,25 @@ public void close() {
     // same executor, it will deadlock: the thread will be stuck waiting for connections
     // to start but cannot start the connections.
     // For this reason, we spawn a dedicated thread. Starting subscriber should be rare.
-    new Thread(new Runnable() {
-      @Override
-      public void run() {
-        try {
-          startPollingConnections();
-          notifyStarted();
-        } catch (Throwable t) {
-          notifyFailed(t);
-        }
-      }
-    }).start();
+    new Thread(
+            new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  // startPollingConnections();
+                  startStreamingConnections();
+                  notifyStarted();
+                } catch (Throwable t) {
+                  notifyFailed(t);
+                }
+              }
+            })
+        .start();
   }
 
   @Override
   protected void doStop() {
-    // stopAllStreamingConnections();
+    stopAllStreamingConnections();
     stopAllPollingConnections();
```

Comment:
It becomes a noop if polling never started. Commenting it out is a good idea though.